src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs

```diff
@@ -704,6 +706,10 @@ private static KeyKind IsKeyProperty(PropertyInfo propertyInfo, KeyAttribute dat
             {
                 keyKind = KeyKind.AttributedKey;
             }
+            else if (propertyInfo.GetCustomAttributes()?.OfType<cm.KeyAttribute>().FirstOrDefault() != null)

```

Comment:
@ElizabethOkerio I could be wrong but I think `GetCustomAttributes` will at worst return an empty collection so `?` might be redundant here

Reply:
Fixed